hadoop-hdfs-project/hadoop-hdfs-httpfs/src/test/java/org/apache/hadoop/fs/http/client/BaseTestHttpFSWith.java

```diff
@@ -2149,6 +2152,44 @@ private void testGetAllEEPolicies() throws Exception {
     }
   }
 
+  private void testGetECCodecs() throws Exception {
+    if (isLocalFS()) {
+      // do not test the testGetECCodecs for local FS.
+      return;
+    }
+    final Path path = new Path("/foo");
+    FileSystem fs = FileSystem.get(path.toUri(), this.getProxiedFSConf());
+    if (fs instanceof DistributedFileSystem) {
+      DistributedFileSystem dfs =
+          (DistributedFileSystem) FileSystem.get(path.toUri(), this.getProxiedFSConf());
+      FileSystem httpFs = this.getHttpFSFileSystem();
+
+      Map<String, String> dfsAllErasureCodingCodecs = dfs.getAllErasureCodingCodecs();
+      Map<String, String> diffErasureCodingCodecs = null;
+
+      if (httpFs instanceof HttpFSFileSystem) {
+        HttpFSFileSystem httpFS = (HttpFSFileSystem) httpFs;
+        diffErasureCodingCodecs = httpFS.getAllErasureCodingCodecs();
+      } else if (httpFs instanceof WebHdfsFileSystem) {
+        WebHdfsFileSystem webHdfsFileSystem = (WebHdfsFileSystem) httpFs;
+        diffErasureCodingCodecs = webHdfsFileSystem.getAllErasureCodingCodecs();
+      } else {
+        Assert.fail(fs.getClass().getSimpleName() +
```

Comment:
use LambdaTestUtils.intercept

Reply:
Thanks for your suggestion, I'll make the required code changes promptly